logstash-core/lib/logstash/runner.rb

```diff
@@ -102,18 +103,16 @@ def execute
 
     return start_shell(@ruby_shell, binding) if @ruby_shell
 
-    if @config_string.nil? && @config_path.nil?
-      fail(I18n.t("logstash.runner.missing-configuration"))
-    end
-
-    @agent.logger = @logger
-
-    config_string = format_config(@config_path, @config_string)
-
-    @agent.add_pipeline("base", config_string, :filter_workers => filter_workers)
+    @agent = create_agent
 
     if config_test?
-      puts "Configuration OK"
+      config_error = @agent.config_valid?
+      if config_error
+        @logger.fatal I18n.t("logstash.error", :error => config_error)
+        return 1
+      else
+        puts "Configuration OK"
```

Comment:
Hmmm, well, this was in the original code. I'm not sure refactoring that is in scope here.